ui/pages/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -594,10 +630,31 @@ export default class ConfirmTransactionBase extends Component {
       addToAddressBookIfNew,
       toAccounts,
       toAddress,
+      ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+      accountType,
+      isNoteToTraderSupported,
+      ///: END:ONLY_INCLUDE_IN
     } = this.props;
-    const { submitting } = this.state;
+    const {
+      submitting,
+      ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+      noteText,
+      ///: END:ONLY_INCLUDE_IN
+    } = this.state;
     const { name } = methodData;
 
+    ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+    if (accountType === 'custody') {
+      txData.custodyStatus = 'created';
+
+      if (isNoteToTraderSupported) {
+        txData.metadata = {
+          note: noteText,
+        };
+      }
+    }
+    ///: END:ONLY_INCLUDE_IN
+
```

Comment:
The code here and also in container component can be moved to component `ConfirmHexData`. It is functional component and selectors can be used to get data from state.

Reply:
@jpuri Not sure I'm following, because in there we want to add a metadata obj to the txData, in order to be send a few [lines bellow here](https://github.com/MetaMask/metamask-extension/blob/4eaee49d1ceee8ff4d7d5a00d9fe97c88b192a26/ui/pages/confirm-transaction-base/confirm-transaction-base.component.js#L711) at `sendTransaction(txData)`

Reply:
Hey, sorry I has confusion that this was part of renderHexData. I agree that is way to do it.